libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -346,6 +346,35 @@ string YulUtilFunctions::overflowCheckedUIntAddFunction(size_t _bits)
 	});
 }
 
+string YulUtilFunctions::overflowCheckedUIntMulFunction(size_t _bits)
+{
+	solAssert(0 < _bits && _bits <= 256 && _bits % 8 == 0, "");
+	string functionName = "checked_mul_uint_" + to_string(_bits);
+	return m_functionCollector->createFunction(functionName, [&]() {
+		return
+			Whiskers(R"(
+			function <functionName>(x, y) -> product {
+				switch x
+				case 0 { product := 0 }
+				default {
+					if lt(div(<mask>, x), y) { revert(0, 0) }
+					<?shortType>
+						product := mulmod(x, y, <powerOfTwo>)
+					<!shortType>
+						product := mul(x, y)
+					</shortType>
+//					if iszero(eq(div(product, x), y)) { revert(0, 0) }
```

Comment:
If this is commented in, then the ``if`` above can be commented out... which is better :-)?

Reply:
I do not understand the above, but I do understand this one...

Reply:
Ok, now I do unterstand the first. The advantage of the first is that it is a condition on the inputs.

Reply:
Yep, that's why it's the one commented in right now - but it's two constants (although ``mask == sub(powerOfTwo,1)``...)... I'm not sure...

Reply:
Both pass the tests in any case.